### PR TITLE
feat(app)!: add group module

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #378 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -360,6 +366,9 @@ func NewRegenApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest
 	)
 	app.AuthzKeeper = authzKeeper
 
+	groupConfig := group.DefaultConfig()
```
Comment (aleem1314) [inline app/app.go:369]: group module [default-config](https://github.com/cosmos/cosmos-sdk/blob/main/x/group/config.go#L17-L18).

File: app/app_test.go [hunk 2]
```diff
@@ -40,3 +65,134 @@ func TestGetMaccPerms(t *testing.T) {
 	dup := GetMaccPerms()
 	require.Equal(t, maccPerms, dup, "duplicated module account permissions differed from actual module account permissions")
 }
+
+func TestRunMigrations(t *testing.T) {
```
Comment (ryanchristo) [inline app/app_test.go:69]: If this is already testing in the sdk, do we need to test in regen ledger? Are there any significant differences?

**Review by ryanchristo — COMMENTED** [review]
Looking good. One question about the added migration test.

File: app/app.go (hunk 1 above)
Comment (ryanchristo) [reply app/app.go:369]: Defaults look good to me.

File: app/app_test.go (hunk 2 above)
Comment (aleem1314) [reply app/app_test.go:69]: Yeah, not required.

**Review by ryanchristo — APPROVED** [review]
LGTM

**Review by technicallyty — APPROVED** [review]
lgtm

**Pull request merged**